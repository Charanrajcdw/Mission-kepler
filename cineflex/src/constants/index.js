export * from "./url.constants";
export * from "./route.constants";
export * from "./component.constants";
export * from "./container.constants";
