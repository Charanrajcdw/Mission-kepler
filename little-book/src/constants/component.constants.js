export const MODAL = {
  members: "Members",
  newBlog: "Add New Blog",
  remove: "Remove",
};

export const MEMBERS_LIST = {
  title: "Members",
  noMembers: "No members available",
};

export const BLOG_FORM = {
  title: "Add New Blog",
  button: "ADD",
  titlePlaceholder: "Name your blog",
  detailsPlaceholder: "Write Content Here ..",
  invalidValues: "Invalid title or description",
  invalidBlog: "Blog title already exists",
  blogAdded: "New Blog added successfully",
  localBlog: "local",
};

export const WARNING_MODAL = {
  title: "Unsaved changes will be lost. Are you sure to continue?",
  continue: "Continue",
  cancel: "Cancel",
};
