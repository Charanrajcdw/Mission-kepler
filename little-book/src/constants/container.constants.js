export const SIDEBAR = {
  logo: "Little Book",
  filter: "FILTER",
  noFilter: "Add blogs to show filters",
  viewMembers: "View Members",
  switchToDark: "Switch to Dark Mode",
  switchToLight: "Switch to Light Mode",
  blogs: "Blogs",
};
