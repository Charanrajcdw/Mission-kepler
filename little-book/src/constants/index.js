export * from "./url.constants";
export * from "./theme.constants";
