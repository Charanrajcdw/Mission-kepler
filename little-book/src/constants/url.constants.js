export const API_URLS = {
  blogs: "https://jsonmockserver.vercel.app/api/blogs",
  users: "https://jsonmockserver.vercel.app/api/users",
  images: "https://jsonmockserver.vercel.app/",
};
