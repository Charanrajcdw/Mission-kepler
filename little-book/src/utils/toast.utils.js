import { toast } from "react-toastify";

export const showSuccessToast = (message, currentTheme) => {
  toast.success(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: currentTheme,
  });
};
