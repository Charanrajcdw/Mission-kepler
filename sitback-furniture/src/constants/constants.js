export const NAVBAR = {
  logo: "SITBACK",
  name: "Nijin Vinodan",
  navlinks: ["couches", "chairs", "dining"],
};

export const BUTTON = {
  shopNow: "SHOP NOW",
  wishlist: "ADD TO WISHLIST",
  cart: "ADD TO CART",
};

export const CATEGORIES = {
  title: "Your Home, With Love",
  description: "Come, Choose from millions of products",
  noCategories: "No categories available now. Pls, check again later",
};

export const PRODUCTS = {
  quantity: "Quantity :",
  noProducts: "No products available now. Pls, check again later",
};

export const ORDERS = {
  title: 'Order Confirmation',
  description: 'Thank you for shopping with us. The items will be delivered withing 7 days.',
  noOrders: 'You have not made any order yet.'
};

export const PAGE_NOT_FOUND = "This page is not available. Click below button to redirect to home page";

export const COPYRIGHTS = "COPYRIGHTS 2020";

export const ROUTES = {
  API_URL: "https://jsonmockserver.vercel.app/api/shopping/furniture",
  categories: "/categories",
  products: "/products",
};
