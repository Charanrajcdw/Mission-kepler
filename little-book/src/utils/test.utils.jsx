import { configureStore } from "@reduxjs/toolkit";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import ThemeContextProvider from "../contexts/theme.context";
import blogSlice from "../store/slices/blogSlice";
import { THEME } from "../constants";

const customRender = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
    ...options
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const customCreateSnapshot = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return renderer.create(<ProviderWrapper>{ui}</ProviderWrapper>).toJSON();
};

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render, customCreateSnapshot as createSnapshot };
