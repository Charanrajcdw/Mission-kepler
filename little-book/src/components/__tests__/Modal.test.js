import { userEvent, render, screen, createSnapshot } from "../../utils/test.utils.jsx";
import { Modal } from "..";
import { MODAL } from "../../constants";

describe("Modal", () => {
  it("should render Modal", () => {
    render(<Modal modalHandler={() => {}}>Modal Content</Modal>);
    const modalElement = screen.getByText("Modal Content");
    const modalContainerElement = modalElement.parentElement;
    expect(modalElement).toBeInTheDocument();
    expect(modalContainerElement).toBeInTheDocument();
  });

  it("should match modal snapshot", () => {
    const modal = createSnapshot(<Modal modalHandler={() => {}}>Modal Content</Modal>);
    expect(modal).toMatchSnapshot();
  });

  it("should call modal close handler on click events if there is no editing", async () => {
    const user = userEvent.setup();
    const modalFn = jest.fn();
    render(<Modal modalHandler={modalFn}>Modal Content</Modal>);
    const modalElement = screen.getByText("Modal Content");
    const modalContainerElement = modalElement.parentElement;
    await user.click(modalElement);
    expect(modalFn).toHaveBeenCalledTimes(0);
    await user.click(modalContainerElement);
    expect(modalFn).toHaveBeenCalledWith(MODAL.remove, false);
  });

  it("should call warning modal handler on click events if there is editing", async () => {
    const user = userEvent.setup();
    const modalFn = jest.fn();
    const initialBlogsData = { isEditing: true };
    render(<Modal modalHandler={modalFn}>Modal Content</Modal>, { preloadedState: { blogs: initialBlogsData } });
    const modalElement = screen.getByText("Modal Content");
    const modalContainerElement = modalElement.parentElement;
    await user.click(modalElement);
    expect(modalFn).toHaveBeenCalledTimes(0);
    await user.click(modalContainerElement);
    expect(modalFn).toHaveBeenCalled();
  });
});
