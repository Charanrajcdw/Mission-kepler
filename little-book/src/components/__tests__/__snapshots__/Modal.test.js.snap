// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match modal snapshot 1`] = `
<div
  className="modalContainer"
  onClick={[Function]}
>
  <div
    className="modalContent"
    onClick={[Function]}
  >
    Modal Content
  </div>
</div>
`;
